src/kbmod/fake_data_creator.py

```diff
@@ -0,0 +1,206 @@
+import os
+import sys
+import random
+
+from astropy.io import fits
+from pathlib import Path
+
+from kbmod.search import *
+
+
+class FakeDataSet:
+    """This class creates fake data sets for testing and demo notebooks."""
+
+    def __init__(self, width, height, num_times, noise_level=2.0, use_seed=False):
+        """The constructor.
+
+        Parameters
+        ----------
+        width : int
+            The width of the images in pixels.
+        height : int
+            The height of the images in pixels.
+        num_times : int
+            The number of times steps (number of images).
+        noise_level : float
+            The level of the background noise.
+        use_seed : bool
+            Use a deterministic seed to avoid flaky tests.
+        """
+        self.width = width
+        self.height = height
+        self.psf_val = 0.5
+        self.noise_level = noise_level
+        self.trajectories = []
+
+        # Generate times with multiple observations per night
+        # separated by ~15 minutes.
+        self.times = []
+        seen_on_day = 0
+        day_num = 0
+        for i in range(num_times):
```

Comment:
This seems like pretty specific logic, particularly the conditional on line 46. Maybe one more comment about why you're resetting `seen_on_day` every three days would be good.

Also is `3` a magic number? Should it be a "global variable" defined outside of `FakeDataSet`? Similarly, is `57130.2` a magic number?

Finally, is there a reason that you use `num_times` and `use_seed` directly, instead of assigning it to a class variable? It looks like `use_seed` is passed to `make_fake_image_stack`, but you could just reference `self.use_seed` instead.

Reply:
Good calls. I made `use_seed` and `num_times` class variables. I also made the PSF value and the number of times to see an object each day arguments.

57130.2 is an arbitrary date. But KBMOD doesn't use absolute dates (just relative dates from the start of the images), so this shouldn't need to be changed.